src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -493,4 +511,56 @@ public static Object coerceBodyType(InputStream body, ResponseBodyType as,
 
         return response;
     }
+
+    private static Timer.Context startTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
+                    requestLine.getMethod());
```

Comment:
We will also want to be careful about the hierarchy for anything we include besides the URI/metric id.  Having the request method reflected in the metrics will be useful for sure, but we might want to think about putting it *before* the URI, in case we think of other sub-uri metrics we want to add on later.

Also, we will probably want to break these metrics into multiple timers so that we can request timing data at any level in the hierarchy; we can look at the metrics stuff in the 'PuppetProfiler' in pe-pse for examples and possibly some re-usable code.